src/CallbackValidator.php

```diff
@@ -25,21 +25,38 @@ class CallbackValidator extends BaseValidator
     /** @var callable Validation callback */
     protected $callback;
 
+    /** @var bool Whether to cache the callback's result */
+    protected $cacheResult;
+
+    /** @var ?bool The result, if {@see self::$cacheResult} is `true` (default) */
+    protected $result;
+
     /**
      * Create a new callback validator
      *
      * @param callable $callback Validation callback
+     * @param boolean $cacheResult Whether to cache the callback's result
      */
-    public function __construct(callable $callback)
+    public function __construct(callable $callback, bool $cacheResult = true)
     {
         $this->callback = $callback;
+        $this->cacheResult = $cacheResult;
     }
 
     public function isValid($value)
     {
+        if ($this->result !== null) {
+            return $this->result;
+        }
+
         // Multiple isValid() calls must not stack validation messages
         $this->clearMessages();
 
-        return call_user_func($this->callback, $value, $this);
+        $valid = call_user_func($this->callback, $value, $this);
```

Comment:
Since we're not enforcing a return type, I would cast to bool since a callback might return null. Or would you say bad luck if you use it wrong?

Reply:
No, you're right.